lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,70 @@
+import { getManagerConfig } from '../../../config';
+import type { RegExManager, RenovateConfig } from '../../../config/types';
+import { getManagerList } from '../../../modules/manager';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig {
+  managerList: Set<string>;
+  managers: WorkerExtractConfig[];
+}
+
+function getFilteredManagerConfig(
+  config: RenovateConfig,
+  manager: string,
+  regexManagerConfig?: RegExManager
+): WorkerExtractConfig {
+  let filteredConfig = {} as WorkerExtractConfig;
+
+  // need to merge the fileMatch config option
+  const fileMatch = [
+    ...(regexManagerConfig?.fileMatch ?? []),
+    ...(config.fileMatch ?? []),
+  ];
+
+  filteredConfig = {
+    ...(regexManagerConfig ?? {}),
+    manager,
+    fileMatch,
+    npmrc: config.npmrc,
+    npmrcMerge: config.npmrcMerge,
+    enabled: config.enabled,
+    ignorePaths: config.ignorePaths ?? [],
+    includePaths: config.includePaths ?? [],
+    skipInstalls: config.skipInstalls,
+    registryAliases: config.registryAliases,
+    fileList: [],
+  };
```

Comment:
Separate out filtering to its own function

Reply:
Do you mean something like this?
 - Divide filetring different manager fields to their own functions....
```ts
getRegexConfig(config){ 
// only return regexManager fields
}

getExtractManagerConfig(config){
// only return extractManager fields and other necessary fields like 
// manager, enabled, fileMatch, ignorePaths, ignoreDeps
}

getFilteredConfig(config){
return {
...(config.manager === 'regex' ? getRegexConfig(config) : {}),
...getExtractManagerConfig(config)
}
}
```